src/backend/distributed/commands/alter_table.c

```diff
@@ -792,14 +789,21 @@ ConvertTable(TableConversionState *con)
 		ExecuteQueryViaSPI(tableConstructionSQL, SPI_OK_UTILITY);
 	}
 
-	char *attachPartitionCommand = NULL;
-	foreach_ptr(attachPartitionCommand, attachPartitionCommands)
+	if (isPartitionedTable)
 	{
-		Node *parseTree = ParseTreeNode(attachPartitionCommand);
+		List *partitionList = PartitionList(con->relationId);
 
-		ProcessUtilityParseTree(parseTree, attachPartitionCommand,
-								PROCESS_UTILITY_QUERY,
-								NULL, None_Receiver, NULL);
+		Oid partitionRelationId = InvalidOid;
+		foreach_oid(partitionRelationId, partitionList)
+		{
+			char *attachPartitionCommand =
+				GenerateAlterTableAttachPartitionCommand(partitionRelationId);
+			Node *parseTree = ParseTreeNode(attachPartitionCommand);
```

Comment:
I wonder how I should free the memory here. I can think of 2 options:
- Using `pfree` in a loop
- Creating a memory context and resetting it.

Reply:
In general we prefer the memory context approach, because that makes it impossible to forget a to free something.